nixos/modules/services/web-apps/immich.nix

```diff
@@ -281,6 +280,9 @@ in
         RuntimeDirectory = "immich";
         User = cfg.user;
         Group = cfg.group;
+        SupplementaryGroups =
+          mkIf (cfg.redis.enable && isRedisUnixSocket)
+          [ config.services.redis.servers.immich.user ];
```

Comment:
Wouldn't this need to be the group rather than the user?

Reply:
The redis module uses `user` for both the user and the group - it doesn't expose the group directly, either to set it or to read it.

https://github.com/NixOS/nixpkgs/blob/1925c603f17fc89f4c8f6bf6f631a802ad85d784/nixos/modules/services/databases/redis.nix#L355-L356

Reply:
I don't like relying on implementation details like this but I don't see an obvious alternative. Please at least add a comment explaining this so that in case this ever breaks future us can quickly figure out why.

Reply:
I see two alternatives:
- use TCP on localhost rather than a unix socket. this is less efficient, but plausible.
- add a group option to the redis module. this seems like a good idea but might be tricky to get through. this does seem worth doing, though.

Reply:
Comment added. I'll put up a PR that adds a "group" option to the redis module, too.